<a href="https://colab.research.google.com/github/Trexroy1010/Machiene-Learning-Projects/blob/main/intover_prediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
#we will use, rfc, svm and simple ann for this,

In [ ]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All"
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/playground-series-s5e7/sample_submission.csv
/kaggle/input/playground-series-s5e7/train.csv
/kaggle/input/playground-series-s5e7/test.csv


In [ ]:
df = pd.read_csv('/kaggle/input/playground-series-s5e7/train.csv')
df.head()

/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1458: RuntimeWarning: invalid value encountered in greater
  has_large_values = (abs_vals > 1e6).any()
/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1459: RuntimeWarning: invalid value encountered in less
  has_small_values = ((abs_vals < 10 ** (-self.digits)) & (abs_vals > 0)).any()
/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1459: RuntimeWarning: invalid value encountered in greater
  has_small_values = ((abs_vals < 10 ** (-self.digits)) & (abs_vals > 0)).any()


,id,Time_spent_Alone,Stage_fear,Social_event_attendance,Going_outside,Drained_after_socializing,Friends_circle_size,Post_frequency,Personality
0,0,0.0,No,6.0,4.0,No,15.0,5.0,Extrovert
1,1,1.0,No,7.0,3.0,No,10.0,8.0,Extrovert
2,2,6.0,Yes,1.0,0.0,NaN,3.0,0.0,Introvert
3,3,3.0,No,7.0,3.0,No,11.0,5.0,Extrovert
4,4,1.0,No,4.0,4.0,No,13.0,NaN,Extrovert


In [ ]:
df.info

/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1458: RuntimeWarning: invalid value encountered in greater
  has_large_values = (abs_vals > 1e6).any()
/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1459: RuntimeWarning: invalid value encountered in less
  has_small_values = ((abs_vals < 10 ** (-self.digits)) & (abs_vals > 0)).any()
/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1459: RuntimeWarning: invalid value encountered in greater
  has_small_values = ((abs_vals < 10 ** (-self.digits)) & (abs_vals > 0)).any()


<bound method DataFrame.info of           id  Time_spent_Alone Stage_fear  Social_event_attendance  \
0          0               0.0         No                      6.0   
1          1               1.0         No                      7.0   
2          2               6.0        Yes                      1.0   
3          3               3.0         No                      7.0   
4          4               1.0         No                      4.0   
...      ...               ...        ...                      ...   
18519  18519               3.0         No                      7.0   
18520  18520               1.0        NaN                      6.0   
18521  18521               7.0        Yes                      1.0   
18522  18522               NaN        Yes                      1.0   
18523  18523               1.0         No                      8.0   

       Going_outside Drained_after_socializing  Friends_circle_size  \
0                4.0                        No          

In [ ]:
df.isnull().sum

<bound method DataFrame.sum of           id  Time_spent_Alone  Stage_fear  Social_event_attendance  \
0      False             False       False                    False   
1      False             False       False                    False   
2      False             False       False                    False   
3      False             False       False                    False   
4      False             False       False                    False   
...      ...               ...         ...                      ...   
18519  False             False       False                    False   
18520  False             False        True                    False   
18521  False             False       False                    False   
18522  False              True       False                    False   
18523  False             False       False                    False   

       Going_outside  Drained_after_socializing  Friends_circle_size  \
0              False                      Fa

In [ ]:
df.dropna()

,id,Time_spent_Alone,Stage_fear,Social_event_attendance,Going_outside,Drained_after_socializing,Friends_circle_size,Post_frequency,Personality
0,0,0.0,No,6.0,4.0,No,15.0,5.0,Extrovert
1,1,1.0,No,7.0,3.0,No,10.0,8.0,Extrovert
3,3,3.0,No,7.0,3.0,No,11.0,5.0,Extrovert
7,7,2.0,No,8.0,3.0,No,4.0,5.0,Extrovert
9,9,1.0,No,8.0,6.0,No,14.0,9.0,Extrovert
...,...,...,...,...,...,...,...,...,...
18509,18509,1.0,No,3.0,4.0,No,15.0,4.0,Extrovert
18511,18511,0.0,No,4.0,5.0,No,11.0,4.0,Extrovert
18514,18514,6.0,No,5.0,3.0,No,10.0,4.0,Extrovert
18519,18519,3.0,No,7.0,3.0,No,9.0,7.0,Extrovert


In [ ]:
df_encoded = pd.get_dummies(data = df, columns = ['Stage_fear', 'Drained_after_socializing', 'Personality'], dtype = int)

In [ ]:
df_encoded.head()

/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1458: RuntimeWarning: invalid value encountered in greater
  has_large_values = (abs_vals > 1e6).any()
/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1459: RuntimeWarning: invalid value encountered in less
  has_small_values = ((abs_vals < 10 ** (-self.digits)) & (abs_vals > 0)).any()
/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1459: RuntimeWarning: invalid value encountered in greater
  has_small_values = ((abs_vals < 10 ** (-self.digits)) & (abs_vals > 0)).any()


,id,Time_spent_Alone,Social_event_attendance,Going_outside,Friends_circle_size,Post_frequency,Stage_fear_No,Stage_fear_Yes,Drained_after_socializing_No,Drained_after_socializing_Yes,Personality_Extrovert,Personality_Introvert
0,0,0.0,6.0,4.0,15.0,5.0,1,0,1,0,1,0
1,1,1.0,7.0,3.0,10.0,8.0,1,0,1,0,1,0
2,2,6.0,1.0,0.0,3.0,0.0,0,1,0,0,0,1
3,3,3.0,7.0,3.0,11.0,5.0,1,0,1,0,1,0
4,4,1.0,4.0,4.0,13.0,NaN,1,0,1,0,1,0


In [ ]:
from sklearn.model_selection import train_test_split

In [ ]:
df_encoded.columns

Index(['id', 'Time_spent_Alone', 'Social_event_attendance', 'Going_outside',
       'Friends_circle_size', 'Post_frequency', 'Stage_fear_No',
       'Stage_fear_Yes', 'Drained_after_socializing_No',
       'Drained_after_socializing_Yes', 'Personality_Extrovert',
       'Personality_Introvert'],
      dtype='object')

In [ ]:
df_encoded.dropna(inplace = True)

In [ ]:
X = df_encoded[['id', 'Time_spent_Alone','Social_event_attendance','Going_outside', 'Friends_circle_size', 'Post_frequency', 'Stage_fear_Yes', 'Drained_after_socializing_Yes']]

In [ ]:
y = df_encoded['Personality_Extrovert']

In [ ]:
X_train, X_test, y_train, y_test = train_test_split(X,y, test_size = 0.3, random_state = 43)

In [ ]:
from sklearn.ensemble import RandomForestClassifier

In [ ]:
model = RandomForestClassifier(n_estimators = 100, random_state = 42)

In [ ]:
model.fit(X_train, y_train)

RandomForestClassifier(random_state=42)

In [ ]:
y_pred = model.predict(X_test)

In [ ]:
from sklearn.metrics import mean_squared_error, r2_score
mse = mean_squared_error(y_test, y_pred)
r2 = r2_score(y_test, y_pred)

In [ ]:
print(f"MSE : {mse}")
print(f"R2 : {r2}")

MSE : 0.03142857142857143
R2 : 0.8182306981564375


In [ ]:
from sklearn.svm import SVC

In [ ]:
model2 = SVC()

In [ ]:
model2.fit(X_train, y_train)

SVC()

In [ ]:
y_pred2 = model.predict(X_test)

In [ ]:
from sklearn.metrics import accuracy_score

In [ ]:
score = accuracy_score(y_test, y_pred2)

In [ ]:
print(score)

0.9685714285714285


AS OF NOW, SVC gave us 96.8%, and RANDOM FOREST HAS GIVEN US = 81.82%, next i will try ANN and see what happens


In [ ]:
import tensorflow as tf

2025-08-13 18:45:34.372309: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:477] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
E0000 00:00:1755110734.589571      36 cuda_dnn.cc:8310] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
E0000 00:00:1755110734.654122      36 cuda_blas.cc:1418] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered


In [ ]:
from sklearn.preprocessing import StandardScaler

In [ ]:
sc = StandardScaler()

In [ ]:
X_train = sc.fit_transform(X_train)

In [ ]:
X_test = sc.fit_transform(X_test)

In [ ]:
X_train.shape

(8981, 8)

In [ ]:
X_train

array([[-0.30600696,  0.00304514,  1.69944015, ...,  1.01250967,
        -0.47547354, -0.47688113],
       [-1.25720931,  2.05417746, -1.22094674, ..., -1.4395665 ,
        -0.47547354,  2.09695865],
       [-1.45607151,  1.0286113 , -1.5859951 , ..., -0.38867671,
         2.10316646,  2.09695865],
       ...,
       [ 0.42815257,  0.68675591, -1.5859951 , ..., -1.0892699 ,
         2.10316646,  2.09695865],
       [-1.11120209, -0.68066563,  0.60429507, ..., -0.03838012,
        -0.47547354, -0.47688113],
       [-0.81206535,  0.00304514,  0.96934343, ..., -0.73897331,
        -0.47547354, -0.47688113]])

In [ ]:
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, LeakyReLU, PReLU, ELU, ReLU, Dropout

In [ ]:
classifier = Sequential()

In [ ]:
classifier.add(Dense(units = 12, activation = 'relu', input_shape = (X_train.shape[1],)))

/usr/local/lib/python3.11/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)
2025-08-13 18:58:11.594892: E external/local_xla/xla/stream_executor/cuda/cuda_driver.cc:152] failed call to cuInit: INTERNAL: CUDA error: Failed call to cuInit: UNKNOWN ERROR (303)


In [ ]:
classifier.add(Dense(units = 7, activation = 'relu'))

In [ ]:
classifier.add(Dense(units = 7, activation = 'relu'))

In [ ]:
classifier.add(Dense(units = 7, activation = 'relu'))

In [ ]:
classifier.add(Dense(units = 7, activation = 'relu'))

In [ ]:
classifier.add(Dense(units = 7, activation = 'relu'))

In [ ]:
#since it's a binary classification, we use sigmoid function
classifier.add(Dense(1, activation = 'sigmoid'))

In [ ]:
classifier.compile(optimizer = 'adam', loss = 'binary_crossentropy', metrics = ['accuracy'])

In [ ]:
model_history = classifier.fit(X_train,y_train, validation_split = 0.33, batch_size = 100, epochs = 100)

Epoch 1/100
61/61 ━━━━━━━━━━━━━━━━━━━━ 4s 9ms/step - accuracy: 0.7647 - loss: 0.6328 - val_accuracy: 0.7578 - val_loss: 0.4215
Epoch 2/100
61/61 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.7702 - loss: 0.3465 - val_accuracy: 0.7578 - val_loss: 0.2840
Epoch 3/100
61/61 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.7660 - loss: 0.2703 - val_accuracy: 0.9464 - val_loss: 0.2696
Epoch 4/100
61/61 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.9611 - loss: 0.2424 - val_accuracy: 0.9639 - val_loss: 0.2609
Epoch 5/100
61/61 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.9651 - loss: 0.2510 - val_accuracy: 0.9636 - val_loss: 0.2529
Epoch 6/100
61/61 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.9635 - loss: 0.2390 - val_accuracy: 0.9649 - val_loss: 0.2457
Epoch 7/100
61/61 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.9646 - loss: 0.2334 - val_accuracy: 0.9642 - val_loss: 0.2396
Epoch 8/100
61/61 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.9662 - loss: 0.2242 - val_accuracy: 0.9636 - v

In [ ]:
model_history.history.keys()

dict_keys(['accuracy', 'loss', 'val_accuracy', 'val_loss'])

In [ ]:
y_pred2 = classifier.predict(X_test)
y_pred2 = (y_pred>0.5)

121/121 ━━━━━━━━━━━━━━━━━━━━ 0s 1ms/step


In [ ]:
from sklearn.metrics import confusion_matrix

In [ ]:
cm = confusion_matrix(y_test, y_pred2)

In [ ]:
cm

array([[ 783,   73],
       [  49, 2945]])

In [ ]:
score2 = accuracy_score(y_pred2, y_test)

In [ ]:
score2

0.9683116883116883

so we got 96.83% accuracy with ann, 0.02 difference with the SVM, we SVM IS better